bot/exts/core/leaderboard.py

```diff
@@ -0,0 +1,206 @@
+import logging
+import math
+from collections import Counter
+
+import discord
+from async_rediscache import RedisCache
+from discord import ButtonStyle, Colour, Interaction, ui
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Roles
+from bot.utils.decorators import with_role
+
+log = logging.getLogger(__name__)
+DUCKY_COINS_THUMBNAIL = (
+    "https://media.discordapp.net/attachments/753252897059373066/902713599884152872/"
+    "new_duck.png?width=230&height=231"
+)
+
+
+class ParentCogConvertor(commands.Converter):
+    """Return the parent cog name for the argument."""
+
+    @staticmethod
+    async def convert(ctx: commands.Context, argument: str) -> str:
+        """Return the parent cog name for the argument."""
+        cog: commands.Cog = ctx.bot.get_cog(argument)
+        if cog:
+            return cog.qualified_name
+
+        cmd: commands.Command = ctx.bot.get_command(argument)
+        if cmd:
+            return cmd.cog_name
+
+        raise commands.BadArgument(
+            f"Unable to convert `{argument}` to valid command or Cog."
+        )
+
+
+class ConfirmClear(ui.View):
+    """A confirmation view for clearing the leaderboard caches."""
+
+    def __init__(self, author_id: int, bot: Bot) -> None:
+        super().__init__(timeout=5)
+        self.confirmed = None
+        self.interaction = None
+        self.authorization = author_id
+        self.bot = bot
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Check the interactor is authorised."""
+        if interaction.user.id == self.authorization:
+            return True
+
+        await interaction.response.send_message(
+            ":no_entry_sign: You are not authorized to perform this action.",
+            ephemeral=True,
+        )
+
+        return False
+
+    @ui.button(label="Confirm", style=ButtonStyle.green, row=0)
+    async def confirm(self, _button: ui.Button, interaction: Interaction) -> None:
+        """Redeploy the specified service."""
+        for global_lb, _ in self.bot.games_leaderboard.values():
+            await global_lb.clear()
+
+        log.info(f"The leaderboard was cleared by Member({self.authorization})")
```

Comment:
```suggestion
        log.info(f"The leaderboard was cleared by {self.authorization}")
```

Discord.py formats this nicely.

Reply:
It is the user ID, discord.py won't format it.